File: src/main/java/com/Hirukar/Project/Models/Beans/Professor.java
package com.Hirukar.Project.Models.Beans;

import com.Hirukar.Project.Models.Beans.Enums.Area;
import java.sql.ResultSet;
import java.sql.SQLException;


public class Professor extends User{
    private String nome;
    private String disciplina;
    private Area area;



    
    Professor(ResultSet rs) throws SQLException {
        super(rs.getString("User.login"),rs.getString("User.senha"),rs.getString("User.email"),rs.getString("User.matricula"));
        this.area = Area.getArea(rs.getInt("User.area"));
    }

    public Professor(String name, String login, String senha, String email, String codigo, Area area) {
        super(login,senha,email,codigo);
    }
    
    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String getDisciplina() {
        return disciplina;
    }

    public void setDisciplina(String disciplina) {
        this.disciplina = disciplina;
    }



    
}

File: src/main/resources/static/js/professor.js

class Professor {
    constructor(codigo,nome,email,telefone) {
        this.codigo = codigo;
        this.nome = nome;
        this.email = email;
        this.telefone = telefone;
    }
    getConteudo(index){
        switch(index){
            case 0: return this.codigo;
            case 1: return this.nome;
            case 2: return this.email;
            case 3: return this.telefone;
        }
    }
    setConteudo(index,conteudo){
        switch(index){
            case 0: this.codigo = conteudo; break;
            case 1: this.nome = conteudo; break;
            case 2: this.email = conteudo; break;
            case 3: this.telefone = conteudo; break;
        }
    }
}